core/cloudflow-operator/src/main/scala/cloudflow/operator/runner/FlinkRunner.scala

```diff
@@ -243,49 +245,55 @@ object FlinkResource {
   }
 
   final case class JobManagerConfig(
-      replicas: Int,
+      replicas: Option[Int],
       resources: Option[Resources] = None,
-      envConfig: EnvConfig = EnvConfig()
+      envConfig: Option[EnvConfig]
   )
 
   final case class TaskManagerConfig(
-      taskSlots: Int,
+      taskSlots: Option[Int],
       resources: Option[Resources] = None,
-      envConfig: EnvConfig = EnvConfig()
+      envConfig: Option[EnvConfig]
```

Comment:
Do these settings need default values for the optional fields (like `resources`) ?

Reply:
Flink operator will actually supply an empty envConfig, so no need to specify a default